src/renderer/components/ft-subscribe-button/ft-subscribe-button.js

```diff
@@ -0,0 +1,143 @@
+import { defineComponent } from 'vue'
+
+import { mapActions } from 'vuex'
+
+import FtButton from '../../components/ft-button/ft-button.vue'
+
+import { MAIN_PROFILE_ID } from '../../../constants'
+import { showToast } from '../../helpers/utils'
+
+export default defineComponent({
+  name: 'FtSubscribeButton',
+  components: {
+    'ft-button': FtButton
+  },
+  props: {
+    channelId: {
+      type: String,
+      required: true
+    },
+    channelName: {
+      type: String,
+      required: true
+    },
+    channelThumbnail: {
+      type: String,
+      required: true
+    },
+    subscriptionCountText: {
+      default: '',
+      type: String,
+      required: false
+    }
+  },
+  computed: {
+    profileList: function () {
+      return this.$store.getters.getProfileList
+    },
+
+    activeProfile: function () {
+      return this.$store.getters.getActiveProfile
+    },
+
+    subscriptionInfo: function () {
+      return this.activeProfile.subscriptions.find((channel) => {
+        return channel.id === this.channelId
+      }) ?? null
+    },
+
+    isSubscribed: function () {
+      return this.subscriptionInfo !== null
+    },
+
+    subscribedText: function () {
+      let subscribedValue = (this.isSubscribed ? this.$t('Channel.Unsubscribe') : this.$t('Channel.Subscribe')).toUpperCase()
+      if (this.subscriptionCountText !== '') {
+        subscribedValue += ' ' + this.subscriptionCountText
+      }
+      return subscribedValue
+    }
+
+  },
+  methods: {
+    handleSubscription: function () {
+      if (this.channelId === '') {
+        return
+      }
+
+      const currentProfile = JSON.parse(JSON.stringify(this.activeProfile))
+      const primaryProfile = JSON.parse(JSON.stringify(this.profileList[0]))
```

Comment:
As this variable is only needed inside the else statement, it could probably be moved down there, that way it would be initialised when unsubscribing.